aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/AWSS3StorageDownloadAccessLevelTest.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage.s3;
+
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.storage.StorageAccessLevel;
+import com.amplifyframework.storage.StorageException;
+import com.amplifyframework.storage.options.StorageDownloadFileOptions;
+import com.amplifyframework.storage.options.StorageUploadFileOptions;
+import com.amplifyframework.storage.result.StorageDownloadFileResult;
+import com.amplifyframework.storage.result.StorageUploadFileResult;
+import com.amplifyframework.testutils.Await;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+
+/**
+ * Instrumentation test to confirm that Storage Download behaves
+ * correctly with regards to the provided storage access level.
+ */
+public final class AWSS3StorageDownloadAccessLevelTest extends StorageInstrumentationTestBase {
+
+    private static final long UPLOAD_SIZE = 100L;
+
+    private final String uploadName = "upload-test-" + System.currentTimeMillis();
+    private File uploadFile;
+
+    private final String destination = "download-test-" + System.currentTimeMillis();
+    private File downloadFile;
+
+    /**
+     * Signs out by default and sets up the file to download object to.
+     * @throws Exception if an error is encountered while creating file
+     */
+    @Before
+    public void setUp() throws Exception {
+        signOut();
+        downloadFile = createTempFile(destination);
+        uploadFile = createTempFile(uploadName, UPLOAD_SIZE);
+    }
+
+    /**
+     * Test downloading with public access without signing in.
+     *
+     * @throws Exception if download is unsuccessful
+     */
+    @Test
+    public void testDownloadUnauthenticatedPublicAccess() throws Exception {
+        final StorageAccessLevel accessLevel = StorageAccessLevel.PUBLIC;
+
+        upload(accessLevel);
```

Comment:
Rather than upload the upload file during each download test (which would result in download tests giving failures if there's only a problem with upload), it'd probably make sense to just upload one file per access level during setup and then clean them up once at the end of all the tests.

Reply:
I agree. I will do the upload portion of the tests before class rather than before each test suite.